File: backend/app/schemas/patient.py
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel


class PatientSummaryResponse(BaseModel):
    id: UUID | str
    source: str
    external_id: str
    name: str | None
    given_name: str | None
    family_name: str | None
    gender: str | None
    birth_date: date | None
    condition_count: int = 0
    medication_count: int = 0
    last_synced_at: datetime | None = None


class PatientPageResponse(BaseModel):
    items: list[PatientSummaryResponse]
    page: int
    page_size: int
    total: int
    has_next: bool


class ConditionResponse(BaseModel):
    id: UUID | str
    external_id: str
    clinical_status: str | None
    verification_status: str | None
    code: str | None
    code_system: str | None
    display: str | None
    onset_date: date | None


class MedicationResponse(BaseModel):
    id: UUID | str
    external_id: str
    status: str | None
    medication_code: str | None
    medication_display: str | None
    authored_on: datetime | None


class PatientDetailsResponse(BaseModel):
    patient: PatientSummaryResponse
    conditions: list[ConditionResponse]
    medications: list[MedicationResponse]

File: ehr-dashboard/frontend/src/types/patient.ts
export interface PatientSummary {
  id: string;
  source: string;
  external_id: string;
  name: string | null;
  given_name: string | null;
  family_name: string | null;
  gender: string | null;
  birth_date: string | null;
  condition_count: number;
  medication_count: number;
  last_synced_at: string | null;
}

export interface Patient extends PatientSummary {}

export interface PatientPage {
  items: PatientSummary[];
  page: number;
  has_next: boolean;
}

export interface Condition {
  id: string;
  external_id: string;
  clinical_status: string | null;
  verification_status: string | null;
  code: string | null;
  code_system: string | null;
  display: string | null;
  onset_date: string | null;
}

export interface Medication {
  id: string;
  external_id: string;
  status: string | null;
  medication_code: string | null;
  medication_display: string | null;
  authored_on: string | null;
}

export interface PatientDetails {
  patient: Patient;
  conditions: Condition[];
  medications: Medication[];
}
